sound/soc/sof/control.c

```diff
@@ -176,12 +181,16 @@ int snd_sof_switch_put(struct snd_kcontrol *kcontrol,
 		update_mute_led(scontrol, kcontrol, ucontrol);
 
 	/* notify DSP of mixer updates */
-	if (pm_runtime_active(scomp->dev))
+	if (pm_runtime_active(scomp->dev)) {
 		snd_sof_ipc_set_get_comp_data(scontrol,
 					      SOF_IPC_COMP_SET_VALUE,
 					      SOF_CTRL_TYPE_VALUE_CHAN_GET,
 					      SOF_CTRL_CMD_SWITCH,
 					      true);
+	} else {
```

Comment:
@mrajwa this is not an error. We did this deliberately to avoid resuming the DSp just to update the control values. When the DSp resumes at some point later on, the control values will be updated then

Reply:
@ranj063 I don't think you understood what this change is about, this is not to stop kcontrols update after resume. It just logs error message and return proper error code when somebody sends kcontrols while the device is not in runtime.

Here is the issue -> The device is not active, somebody from user space sends kcontrols (using sof-ctl for example). Now, they get SUCCESS and absolutely no error message therefore thinking kcontrols has been sent and set while neither of those two is true.

Reply:
@mrajwa I still think this is intentional behavior. The kcontrol has been set successfully (to the cache). In what scenario this is a problem?